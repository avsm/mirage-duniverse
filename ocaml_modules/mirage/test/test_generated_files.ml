let with_fmt_str k =
  let buf = Buffer.create 0 in
  let fmt = Format.formatter_of_buffer buf in
  k fmt;
  Format.pp_print_flush fmt ();
  Buffer.contents buf

let redact_line n s =
  (* At the moment it is not possible to inject argv & time so we remove the
   * generated line.
   * See https://github.com/mirage/functoria/pull/159 *)
  let lines = String.split_on_char '\n' s in
  let redacted_lines =
    List.mapi (fun i line -> if i + 1 = n then "# REDACTED" else line) lines
  in
  String.concat "\n" redacted_lines

let print_banner s =
  print_endline s;
  print_endline @@ String.make (String.length s) '=';
  print_newline ()

let test_output_fat =
  Mirage_impl_fs.fat_shell_script ~block_file:"BLOCK_FILE"
    ~root:(Fpath.v "ROOT") ~dir:(Fpath.v "DIR") ~regexp:"REGEXP"

let () =
  let tests = [("output_fat", test_output_fat)] in
  List.iter
    (fun (name, f) ->
      print_banner name;
      print_endline @@ with_fmt_str f )
    tests
