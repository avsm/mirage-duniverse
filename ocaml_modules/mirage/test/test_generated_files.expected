output_fat
==========

#!/bin/sh

echo This uses the 'fat' command-line tool to build a simple FAT
echo filesystem image.

FAT=$(which fat)
if [ ! -x "${FAT}" ]; then
  echo I couldn\'t find the 'fat' command-line tool.
  echo Try running 'opam install fat-filesystem'
  exit 1
fi

IMG=$(pwd)/BLOCK_FILE
rm -f ${IMG}
cd ROOT/DIR
SIZE=$(du -s . | cut -f 1)
${FAT} create ${IMG} ${SIZE}KiB
${FAT} add ${IMG} REGEXP
echo Created 'BLOCK_FILE'

